source/AI.cpp

```diff
@@ -1712,11 +1727,26 @@ bool AI::ShouldDock(const Ship &ship, const Ship &parent, const System *playerSy
 
 	// TODO: Reboard if in need of ammo.
 
+	// Reboard if low power/no power (battery only).
+	if(ship.IsEnergyLow())
+		return true;
+
+	// Reboard/retreat if harmless and enemy nearby; for example boxwings.
+	if(!ship.IsArmed(true) && ship.IsEnemyInEscortSystem())
+		return true;
+
 	// If a carried ship has fuel capacity but is very low, it should return if
-	// the parent can refuel it.
-	double maxFuel = ship.Attributes().Get("fuel capacity");
-	if(maxFuel && ship.Fuel() < .005 && parent.JumpFuel() < parent.Fuel() *
-			parent.Attributes().Get("fuel capacity") - maxFuel)
+	// the parent can refuel it.  Also account for when fighter has ramscoop to
+	// prioritize returning over ramscoop regeneration.
+	bool readyToRefuelCarrier = ship.IsEscortsFullOfFuel();
+	// Fuel is at least 50% or 100 jump fuel; whichever is lower.
+	bool fighterHasRefueled = ship.CanRefuel(parent);
+	bool parentIsNotFullOfFuel = parent.Fuel() < 1.;
+	//bool shipIsLowFuel = ship.IsFuelLow();
+	// Only return to ship if low fuel or if the fighter has ramscoop and is refueling the carrier.
+	bool shouldReturnForFuel = (readyToRefuelCarrier) ? !(fighterHasRefueled && parentIsNotFullOfFuel) : ship.IsFuelLow() && parent.CanRefuel(ship);
+	// TODO: only dock for fuel if no orders
+	if((shouldReturnForFuel ^ readyToRefuelCarrier) && (!ship.IsYours() || (!orders.count(&ship) && ship.CanRefuel(parent))))
```

Comment:
Xor? Did you mean to use a simple or here? I'm honestly confused why you'd use a xor here (and in a few other places as well).

Reply:
I can also add a code comment for this so it is more obvious.

Reply:
This is intended behavior.  I am resolving this feedback.